clients/vmClient/vmClient.go

```diff
@@ -856,4 +857,23 @@ next:
 	return nil
 }
 
+func verifyInstanceSizeAvailableInLocation(location *locationClient.Location, vmSize string) error {
```

Comment:
I've done this, however it means either an unnecessary API call or removing the list of available sizes from the error message. In this instance I've chosen to remove the list of available sizes.